sotodlib/tod_ops/sub_polyf.py

```diff
@@ -1,8 +1,176 @@
 import numpy as np
 import logging
+from scipy.special import eval_legendre
 from . import flags
 logger = logging.getLogger(__name__)
 
+def _mask_data(_data, mask):
+    data = np.ma.masked_array(_data,mask=mask)
+    return data
+
+def subscan_legendre_filter(aman, degree, signal=None, exclude_turnarounds=False, mask=None, in_place=True):
+    """
+    Apply legendre filtering to subscan segments in a data array.
+    This function applies legendre poly-filtering to subscan segments within signal for each detector.
+    Subscan segments are defined based on the presence of flags such as 'left_scan' and 'right_scan'. Polynomial filtering
+    is used to remove low-degree polynomial trends within each subscan segment.
+
+    Arguments
+    ---------
+    aman : AxisManager
+    degree : int
+        The degree of the polynomial to be removed.
+    signal : array-like, optional
+        The TOD signal to use. If not provided, `aman.signal` will be used.
+    exclude_turnarounds : bool
+        Optional. If True, turnarounds are excluded from subscan identification. Default is False.
+    mask : str or RangesMatrix
+        Optional. A mask used to select specific data points for filtering. Default is None.
+        If None, no mask is applied. If the mask is given in str, ``aman.flags['mask']`` is used as mask.
+        Arbitrary mask can be specified in the style of RangesMatrix.
+    in_place: bool
+        Optional. If True, `aman.signal` is overwritten with the processed signal.
+
+    Returns
+    -------
+    signal : array-like
+        The processed signal.
+    """
+
+    if signal is None:
+        signal = aman.signal
+
+    if not(in_place):
+        signal = signal.copy()
+
+    time = np.copy(aman["timestamps"])
+        
+    if exclude_turnarounds:
```

Comment:
A few tens of lines in the beginning is just copy-and-past of `subscan_polyfilter`, which makes maintenance of code difficult.
Can we merge the two functions into one wrapper function, `subscan_polyfilter`, which has an argument of `method` or `lin_reg` like the hwpss function and refers each mode function?

https://github.com/simonsobs/sotodlib/blob/0ca1d61e7e7b73869b3afb71363c9a7b0c8f01a8/sotodlib/hwp/hwp.py#L34-L36

Reply:
Two functions are merged and argument "method" is added.